### ISSUE TITLE
metacity-decoration edge glitches and other issues with HIDPI

### ISSUE THREAD

**Issue by raveit65** [mate-desktop/marco#496]
#### Expected behaviour
clear and defined edges with all metacity themes if using HIDPI resolutions.
Correct color gradient with Traditional themes

#### Actual behaviour
Edges are undefined with double border lines
 Color gradient is broken with Traditional themes
#### Steps to reproduce the behaviour

Using Traditional, Submarine, GreenLaguna and Menta themes with marco VM with a HIDPI resolution.

#### MATE general version
1.21.1

#### Package version
1.21.x, 1.20.x and master

#### Linux Distribution
fedora 30

#### Link to downstream report of your Distribution
no need to file out a report against myself, i am informed :)

First, all this issues doesn't exists with normal monitor resolutions.
Metacity WM display all this themes well with HIDPI.

TraditionalOK with marco:
![traditionalOK-marco](https://user-images.githubusercontent.com/961604/57582775-0c974700-74c9-11e9-95f8-c664920b6447.png)
TraditionalOK with metacity:
![traditionalOK-metacity](https://user-images.githubusercontent.com/961604/57582798-6c8ded80-74c9-11e9-93a2-1147c65504c4.png)
GreenLaguna with marco:
![greenlaguna-marco](https://user-images.githubusercontent.com/961604/57582803-7b74a000-74c9-11e9-89b8-99e26534a855.png)
GreenLaguna with metacity:
![greenlaguna-metacity](https://user-images.githubusercontent.com/961604/57582808-921af700-74c9-11e9-9369-3933936ca7b8.png)
Blue-Submarine with marco:
![blue-submarine-marco](https://user-images.githubusercontent.com/961604/57582816-b24ab600-74c9-11e9-89a7-ff451194d41f.png)

Menta themes have edge glitches too, but not always. Mostly it happaens if using tiling modes.
I will make screenshots when it happens.

**Comment by vkareh** [comment]
I also experience these issues with the Ubuntu MATE themes.

**Comment by raveit65** [comment]
Using Menta theme in tiling mode.
![menta-marco](https://user-images.githubusercontent.com/961604/57583681-25a4f580-74d3-11e9-9b53-c2d488773a15.png)
You see the double border better on you box as with a screenshot.
Its a very thin line.

**Comment by raveit65** [comment]
I added back a metacity-theme-3.xml for GreenLaguna which fixes the glitches.
https://github.com/mate-desktop/mate-themes/commit/7b4a343d8962fe6477ffbeebfa6a22ded46fa004

**Comment by raveit65** [comment]
Glitches with borders when windows are in tiling mode aren't fixed. Eg. Menta.
But i see that thin double border line with other themes too.

**Comment by vkareh** [comment]
Yes, you're right. The tiling stuff needs to be addressed separately. There's no commit to backport for this, so we'll need to debug on our own.

**Comment by raveit65** [comment]
I think i found the cause of the last remaining glitches when window is in corner-tiling mode.
It happens only when i use a third vertical panel which is my default.
When using only top and bottom panel, or moving the third panel away from monitor edges, than the top right corner of the window looks OK for me.

With third vertical panel: (bad dark line and a step at the edge)
![marco-border-bad](https://user-images.githubusercontent.com/961604/68330591-cfa4b980-00d3-11ea-8c41-43a12d90a2de.png)

Without third vertical panel:
![marco-border-good](https://user-images.githubusercontent.com/961604/68330637-e2b78980-00d3-11ea-9027-910ca1f47e9b.png)

Note : you only see that with HIDPI and a light theme without a dark borders like Menta themes.
With a theme with dark borders, eg. BlueSubmarine yoiu don't see the issue.
Last little glitches in screenshot you don't see on monitor, only here in screenshot.

**Comment by raveit65** [comment]
I opened a new report for last single issue.
https://github.com/mate-desktop/marco/issues/573

**Comment by killerkalamari** [comment]
I'm on MATE 1.20.5 (Debian 10 Buster), and still have the problem. I suspect that's simply because my version is older than the fix, so I just need to upgrade, which I assume will be in Debian 11. Until then, I modified my preferred window decoration, TraditionalOk, by commenting out all the gradients and adding rectangles. It doesn't look exactly the same, but it's close enough. Example from `/usr/share/themes/TraditionalOk/metacity-1/metacity-theme-1.xml`:

```
    <!-- ** fancy gradient ** -->
    <!-- <gradient type="vertical" x="2" y="top_height/2" width="width-4" height="top_height/2-1">
        <color value="shade/#86abd9/1.0"/>
        <color value="shade/#86abd9/0.94"/>
    </gradient> -->
    <rectangle color="shade/#86abd9/1.0" filled="true" x="2" y="top_height/2" width="width-4" height="top_height/2-1"/>
    <!-- <gradient type="vertical" x="2" y="2" width="width-4" height="top_height/2-2">
        <color value="shade/#86abd9/1.08"/>
        <color value="shade/#86abd9/1.02"/>
    </gradient> -->
```

Note that I just used `1.0` for all the shade values to keep it simple. I could probably also have combined many of the gradients into a single rectangle, but keeping it in two parts meant that I could copy and paste the x/y/width/height part of the line straight from the gradient into the rectangle and I didn't have to recalculate anything, so that was easier. 

It's straightforward to make an installable theme based on an existing theme, and that's what I'd recommend, rather than editing any files under `/usr/share/themes`. Copy `/usr/share/themes/TraditionalOk` to your home directory and work on `metacity-theme-1.xml` from there. Also edit `index.theme` and `metacity-theme-1.xml` to change all instances of `TraditionalOk` to `TraditionalOkRectangle`. Finally, compress it to a `.tar.gz`.

I've attached the TraditionalOkRectangle theme that I created as above:
[TraditionalOkRectangle.tar.gz](https://github.com/mate-desktop/marco/files/6819024/TraditionalOkRectangle.tar.gz)
